packages/gator-permissions-snap/src/clients/blockchainMetadataClient.ts

```diff
@@ -102,38 +102,48 @@ export class BlockchainTokenMetadataClient implements TokenMetadataClient {
       };
     }
 
-    const [balanceEncoded, decimalsEncoded, symbolEncoded] = await Promise.all([
-      this.#ethereumProvider.request<Hex>({
-        method: 'eth_call',
-        params: [
-          {
-            to: assetAddress,
-            data: BlockchainTokenMetadataClient.#balanceOfCalldata,
-          },
-          'latest',
-        ],
-      }),
-      this.#ethereumProvider.request<Hex>({
-        method: 'eth_call',
-        params: [
-          {
-            to: assetAddress,
-            data: BlockchainTokenMetadataClient.#decimalsCalldata,
-          },
-          'latest',
-        ],
-      }),
-      this.#ethereumProvider.request<Hex>({
-        method: 'eth_call',
-        params: [
-          {
-            to: assetAddress,
-            data: BlockchainTokenMetadataClient.#symbolCalldata,
-          },
-          'latest',
-        ],
-      }),
-    ]);
+    let balanceEncoded;
+    let decimalsEncoded;
+    let symbolEncoded;
+
+    try {
+      [balanceEncoded, decimalsEncoded, symbolEncoded] = await Promise.all([
+        this.#ethereumProvider.request<Hex>({
+          method: 'eth_call',
+          params: [
+            {
+              to: assetAddress,
+              data:
+                BlockchainTokenMetadataClient.#balanceOfCalldata +
+                account.slice(2).padStart(64, '0'),
+            },
+            'latest',
+          ],
+        }),
+        this.#ethereumProvider.request<Hex>({
+          method: 'eth_call',
+          params: [
+            {
+              to: assetAddress,
+              data: BlockchainTokenMetadataClient.#decimalsCalldata,
+            },
+            'latest',
+          ],
+        }),
+        this.#ethereumProvider.request<Hex>({
+          method: 'eth_call',
+          params: [
+            {
+              to: assetAddress,
+              data: BlockchainTokenMetadataClient.#symbolCalldata,
+            },
+            'latest',
+          ],
+        }),
+      ]);
+    } catch (error) {
+      throw new Error('Failed to fetch token balance and metadata');
```

Comment:
I'm not sure how well it would be surfaced, but `throw new Error('Failed to fetch token balance and metadata', { cause: error });` might be helpful to ensure useful information is surfaced (especially if we start indexing errors).

`logger.error('...');` might also be useful here.

Same comment regarding the other throw errors below.

Reply:
Added `logger.error`. For this I think its sensible to surface only the error that is useful for developer implementing  and logger takes care of the actual underlying error.